src/pages/ViewArticle.jsx

```diff
@@ -0,0 +1,33 @@
+import { useParams } from "react-router-dom";
+import { getArticle } from "../api";
+import { useState, useEffect } from "react";
+
+function ViewArticle() {
+    // id needs to have the same name as the route path! 
+    const { id } = useParams();
+  const [singleArticle, setSingleArticle] = useState({});
+
+
+  console.log(id, "articleid");
+
+  useEffect(() => {
+    getArticle(id).then((articleData) => {
+      setSingleArticle(articleData);
+    });
+  }, [id]);
```

Comment:
This is really good logic, but dont forget to remove console.log's and comments that are no longer needed